akka-stream/src/main/scala/akka/stream/Attributes.scala

```diff
@@ -43,74 +48,37 @@ final case class Attributes(attributeList: List[Attributes.Attribute] = Nil) {
   }
 
   /**
-   * Java API
+   * Java API: Get the most specific attribute value for a given Attribute type or subclass thereof.
+   * If no such attribute exists, return a `default` value.
    *
-   * The list is ordered with the most specific attribute first, least specific last.
-   * Note that the order was the opposite in Akka 2.4.x.
-   */
-  def getAttributeList(): java.util.List[Attribute] = {
-    import scala.collection.JavaConverters._
-    attributeList.asJava
-  }
-
-  /**
-   * Java API: Get all attributes of a given `Class` or
-   * subclass thereof.
+   * The most specific value is the value that was added closest to the graph or stage itself or if
+   * the same attribute was added multiple times to the same graph, the last to be added.
    *
-   * The list is ordered with the most specific attribute first, least specific last.
-   * Note that the order was the opposite in Akka 2.4.x.
-   */
-  def getAttributeList[T <: Attribute](c: Class[T]): java.util.List[T] =
-    if (attributeList.isEmpty) java.util.Collections.emptyList()
-    else {
-      val result = new java.util.ArrayList[T]
-      attributeList.foreach { a ⇒
-        if (c.isInstance(a))
-          result.add(c.cast(a))
-      }
-      result
-    }
-
-  /**
-   * Java API: Get the most specific attribute (added last) of a given `Class` or subclass thereof.
-   * If no such attribute exists the `default` value is returned.
+   * This is the expected way for stages to access attributes.
    */
   def getAttribute[T <: Attribute](c: Class[T], default: T): T =
     getAttribute(c).orElse(default)
 
   /**
-   * Java API: Get the least specific attribute (added first) of a given `Class` or subclass thereof.
-   * If no such attribute exists the `default` value is returned.
-   */
-  def getFirstAttribute[T <: Attribute](c: Class[T], default: T): T =
-    getFirstAttribute(c).orElse(default)
-
-  /**
-   * Java API: Get the most specific attribute (added last) of a given `Class` or subclass thereof.
+   * Java API: Get the most specific attribute value for a given Attribute type or subclass thereof.
+   * If no such attribute exists, return a `default` value.
```

Comment:
The method below does not take `default`. Probably documentation from another method here.